datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -1034,29 +1033,50 @@ fn is_time_with_valid_unit(datatype: DataType) -> bool {
     )
 }
 
+/// Timezone coercion is handled by the following rules:
```

Comment:
Could we add some rationale to this comment about why one might choose strict vs non-strict coercion? 

It looks like nonstrict coercion is used for `values` and `comparison` coercion but not function arguments (which makes sense to me, as the function signature has some way to specify what timezones it will handle):

https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.TypeSignature.html#data-types